lib/python/picongpu/plugins/data/xray_diffraction.py

```diff
@@ -0,0 +1,234 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017-2020 PIConGPU contributors
+Authors: Axel Huebl, Finn-Ole Carstens, Juncheng E, Sergei Bastrakov
+License: GPLv3+
+"""
+from .base_reader import DataReader
+
+import numpy as np
+import os
+
+
+class XrayDiffractionData(DataReader):
+    """
+    Data Reader for X-ray diffraction Plugin.
+    """
+
+    def __init__(self, run_directory):
+        """
+        Parameters
+        ----------
+        run_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        super().__init__(run_directory)
+
+        self.data_file_prefix = "_intensity_"
+        self.data_file_suffix = ".dat"
+        self.data_file_folder = "xrayDiffraction/"
+        self.data = None
+
+        # TODO: update
```

Comment:
Must this todo be solved before we can merge?

Reply:
With this we need to decide. I made a small Python visualization script based on code developed by @ejcjason to reproduce pictures in his PR, and they look alike. Will attach the pictures to this topic tomorrow. However, the Python part currently does not fit our other Python modules to access plugin data.

So I think we should either merge only the C++ part, or I should finish the Python part before merging. Today I asked @PrometheusPi about what plugins to use as examples, and it seems more or less clear, will probably take me a few work hours to prepare it for this PR. I think since we are not including it in 0.5.0, makes sense to do alltogether.